README.md

```diff
@@ -12,15 +12,13 @@ To define a counter widget which increments when clicked, you can define the fol
 
 ```js
 class Counter extends HTMLElement {
-  clicked() {
-    this.x++;
-    window.requestAnimationFrame(this.render.bind(this));
-  }
-
   constructor() {
     super();
-    this.onclick = this.clicked.bind(this);
     this.x = 0;
+    this.onclick = () => {
```

Comment:
The point of this example is to show the current pattern, which is a constructor-bound instance method. Using an arrow function in a class field is not what’s desired here.

Reply:
I used this because, as I understand, this is closer to what happens under the hood when you use a class field method.

Reply:
This section is supposed to be in ES2015 - class fields aren’t supported in ES2015.

The point is to show how you’d want to implement a combo prototype/bound method now, versus how you’d want to do it with class fields - neither has an arrow function in an own property.

Reply:
> This section is supposed to be in ES2015 - class fields aren’t supported in ES2015.

This section remains to be in es2015. Using arrow function in the constructor is valid es2015 code.

> The point is to show how you’d want to implement a combo prototype/bound method now, versus how you’d want to do it with class fields

But currently the code snippets show no difference. You make `x` a class field, but you still bind the class method to the instance in the constructor. So what is the point? Other than to just show new syntax, which is ok.

What I want is to document how you can use class fields with arrow functions and avoid having to create a bound method.

Reply:
oh lol, you're right, sorry :-) i saw it as a class field.

The point is that it's more testable, more performant, and more conceptually correct to put the function on the prototype, and to have the "own property" part be nothing more than a call into that shared method.